enhancements/oc/oc-login-external-oidc-integration.md

```diff
@@ -0,0 +1,244 @@
+| Supported | Yes                                                                        | Yes                                                                                                      | Yes                                                                        | Yes                                                                                                      | No                         | No                                                                       | Yes                                     | No                       | No                                |
+| Comment   |                                                                            | Only if provider supports this[1]                                                                        |                                                                            | Only if provider supports this[1]                                                                        | Deprecated[3]              | Mostly used in CI systems that <br/>client-secret can be stored securely | If provider supports it                 |                          | Deprecated[4]                     |
+| Command   | `oc login api.url --issuer-url=external.oidc.issuer --client-id=client-id` | `oc login api.url --issuer-url=external.oidc.issuer --client-id=client-id --client-secret=client-secret` | `oc login api.url --issuer-url=external.oidc.issuer --client-id=client-id` | `oc login api.url --issuer-url=external.oidc.issuer --client-id=client-id --client-secret=client-secret` |                            |                                                                          | [2]                                     |                          |                                   |
+
+[1] For instance, Azure Entra ID supports configuring client secret to behave as confidential client in Auth Code Authentication Flow. Contrarily, Keycloak does not.
+
+[2] Refresh token does not provide any interface, because it is handled automatically by `oc get-token`, if provider supports it.
+
+[3] https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-10#name-removal-of-the-oauth-20-imp
+
+[4] https://datatracker.ietf.org/doc/html/draft-ietf-oauth-security-topics-24#section-2.4
+
+### API Extensions
+
+N/A
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+### Risks and Mitigations
+
+Additional configuration steps need to be performed in external OIDC side which may have slight negative impact on usability initially.
+For example;
+* User has to open the Keycloak dashboard
+* Creates a client-id by also setting `127.0.0.1` as valid redirect url
+* Enable Auth Code Flow
```

Comment:
Should we summarize this as "An administrator must create an OIDC client within their identity provider so that it can be used by their users"?

Reply:
yes, updated.